### PR TITLE
Switch from TSLint to ESLint

### PR DESCRIPTION
**Pull request by trivikr**
TSLint has been deprecated:
* Blog post from Palantir https://medium.com/palantir/tslint-in-2019-1a144c2317a9
* Issues: https://github.com/palantir/tslint/issues/4534

After moving to eslint, confirmed that:
* `yarn test` runs eslint
* `yarn lint` runs prettier
* `yarn fix` fixes lint/prettier issues
* `yarn lint` is successful

### CODE REVIEW

**Comment by Turbo87** [comment]
looks like one of the dep versions is not compatible with Node 6 though. we should probably drop Node 6 support, but it would have to be in a dedicated PR so that we can correctly label it as "breaking".

**Comment by trivikr** [comment]
How about releasing v1.0.0 of `lerna-changelog` dropping support for Node.js v6?
It's EoL was on 2019-04-30

**Comment by Turbo87** [comment]
@trivikr can you rebase this?

**Comment by trivikr** [comment]
> @trivikr can you rebase this?

@Turbo87 Done!

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "changelog": "node ./bin/cli.js",
     "clean": "rimraf lib",
     "fix": "npm run lint -- --fix",
-    "lint": "tslint --project . --format stylish",
+    "lint": "eslint 'src/**/*.ts' --format stylish",
```
Comment (Turbo87) [inline package.json]: do we not have to set something like `--ext ts`?

```suggestion
    "lint": "eslint src --ext ts --format stylish",
```

Comment (trivikr) [reply package.json]: Difference between the debug outputs:
* `eslint 'src/**/*.ts' --format stylish`: [output-src.txt](https://github.com/lerna/lerna-changelog/files/3833217/output-src.txt)
* `eslint src --ext ts --format stylish`: [output-ext.txt](https://github.com/lerna/lerna-changelog/files/3833239/output-ext.txt)

The out is same, but latter one is readable so going ahead with it
output-diff

**Comment by Turbo87** [comment]
thanks :)

**Pull request merged**